tests/Stripe/StripeClientTest.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Stripe;
+
+class StripeClientTest extends TestCase
+{
+    public function testCtorDoesNotThrowIfApiKeyIsNull()
+    {
+        $client = new StripeClient(null);
+        $this->assertNotNull($client);
+        $this->assertNull($client->getApiKey());
+    }
+
+    /**
+     * @expectedException \Stripe\Exception\InvalidArgumentException
+     * @expectedExceptionMessage API key cannot be the empty string.
+     */
+    public function testCtorThrowsIfApiKeyIsEmpty()
+    {
+        $client = new StripeClient('');
+    }
+
+    /**
+     * @expectedException \Stripe\Exception\InvalidArgumentException
+     * @expectedExceptionMessage API key cannot contain whitespace.
+     */
+    public function testCtorThrowsIfApiKeyContainsWhitespace()
+    {
+        $client = new StripeClient("sk_test_123\n");
+    }
+
+    public function testRequestWithClientApiKey()
+    {
+        $client = new StripeClient("sk_test_123", null, MOCK_URL);
+        $charge = $client->request("get", "/v1/charges/ch_123", [], []);
+        $this->assertNotNull($charge);
+        $this->assertEquals("ch_123", $charge->id);
```

Comment:
the test implies you test the API key, you can look at $charge to confirm it's the right key
But really we shouldn't have the $charge have the API key (people dump it with var_dump).